src/hotspot/share/utilities/xmlstream.hpp

```diff
@@ -109,29 +110,43 @@ class xmlStream : public outputStream {
   int unflushed_count() { return (int)(out()->count() - _last_flush); }
 
   // writing complete XML elements
-  void          elem(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
-  void    begin_elem(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
-  void      end_elem(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
+  ATTRIBUTE_PRINTF(2, 3)
+  void          elem(const char* format, ...);
+  ATTRIBUTE_PRINTF(2, 3)
+  void    begin_elem(const char* format, ...);
+  ATTRIBUTE_PRINTF(2, 3)
+  void      end_elem(const char* format, ...);
   void      end_elem();
-  void          head(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
-  void    begin_head(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
-  void      end_head(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
+  ATTRIBUTE_PRINTF(2, 3)
+  void          head(const char* format, ...);
+  ATTRIBUTE_PRINTF(2, 3)
+  void    begin_head(const char* format, ...);
+  ATTRIBUTE_PRINTF(2, 3)
+  void      end_head(const char* format, ...);
   void      end_head();
-  void          done(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);  // xxx_done event, plus tail
+  ATTRIBUTE_PRINTF(2, 3)
+  void          done(const char* format, ...);  // xxx_done event, plus tail
   void          done_raw(const char * kind);
   void          tail(const char* kind);
 
   // va_list versions
-  void       va_elem(const char* format, va_list ap) ATTRIBUTE_PRINTF(2, 0);
-  void va_begin_elem(const char* format, va_list ap) ATTRIBUTE_PRINTF(2, 0);
-  void       va_head(const char* format, va_list ap) ATTRIBUTE_PRINTF(2, 0);
-  void va_begin_head(const char* format, va_list ap) ATTRIBUTE_PRINTF(2, 0);
-  void       va_done(const char* format, va_list ap) ATTRIBUTE_PRINTF(2, 0);
+  ATTRIBUTE_PRINTF(2, 0)
+  void       va_elem(const char* format, va_list ap);
+  ATTRIBUTE_PRINTF(2, 0)
+  void va_begin_elem(const char* format, va_list ap);
+  ATTRIBUTE_PRINTF(2, 0)
+  void       va_head(const char* format, va_list ap);
+  ATTRIBUTE_PRINTF(2, 0)
+  void va_begin_head(const char* format, va_list ap);
+  ATTRIBUTE_PRINTF(2, 0)
+  void       va_done(const char* format, va_list ap);
 
   // write text (with quoting of special XML characters <>&'" etc.)
   outputStream* text() { return _text; }
-  void          text(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
-  void       va_text(const char* format, va_list ap) ATTRIBUTE_PRINTF(2, 0) {
+  ATTRIBUTE_PRINTF(2, 3)
+  void          text(const char* format, ...);
+  ATTRIBUTE_PRINTF(2, 0)
+  void       va_text(const char* format, va_list ap) {
```

Comment:
This file is a particularly bad (to me) example of what happens without whitespace between a declaration
and the attributes for the next declaration.  I find this really hard to parse.  And the extra whitespace following
return types makes it even worse for me.

Reply:
+1